pkg/reconciler/instances/istio/bootstrap.go

```diff
@@ -101,17 +101,31 @@ func parsePaths(input string, isValid func(string) error) ([]string, error) {
 }
 
 // TODO: test it
-func validatePath(path string) error {
+func validatePath(path string, logger *zap.SugaredLogger) error {
 	stat, err := os.Stat(path)
 	if err != nil {
 		return errors.Wrap(err, "Error getting file data")
 	}
 	mode := stat.Mode()
+	logger.Debugf("%s mode: %s", path, mode)
 	if (!mode.IsRegular()) || mode.IsDir() {
 		return errors.New(fmt.Sprintf("\"%s\" is not a regular file", path))
 	}
 	if uint32(mode&0111) == 0 {
-		return errors.New(fmt.Sprintf("\"%s\" is not executable", path))
+		logger.Debugf("%s is not executable, will chmod +x", path)
+		var fileMode os.FileMode = 0777
+		if err := os.Chmod(path, fileMode); err != nil {
```

Comment:
I believe this logic should not be a part of this function. It was designed to check whether path exists and is executable

Reply:
I can create a function which will then be called there. We need it here, since to persist file permission in Windows is very weirdly behaving. I tested it the full day yesterday on a local Windows VM , and that is the only way I got it to work . But yes, I will move it into a separate function.

Reply:
done